dsc/workflows/base/simple_csv.py

```diff
@@ -20,6 +22,84 @@ class SimpleCSV(Workflow):
 
     workflow_name: str = "simple_csv"
 
+    def reconcile_bitstreams_and_metadata(
+        self, metadata_file: str = "metadata.csv"
+    ) -> None:
+        """Summary.
+
+        Args:
+            metadata_file (str, optional): _description_. Defaults to "metadata.csv".
+        """
+        # get item identifiers from bitstreams and metadata file
+        bitstream_item_identifiers = self._get_item_identifiers_from_bitstreams(
+            metadata_file
+        )
+        metadata_item_identifiers = self._get_item_identifiers_from_metadata(
+            metadata_file
+        )
+
+        # get matching item identifiers (IDs for items with both bitstreams and metadata)
+        matching_item_identifiers = bitstream_item_identifiers & metadata_item_identifiers
```

Comment:
Wild!  I've never seen or used the `&` operator for two sets ([documentation](https://docs.python.org/3/library/stdtypes.html#frozenset.intersection)).

I think sets are and should be central in lots of work we do, so in full support of getting familiar with all the syntaxes and approaches.

Reply:
Spent a good chunk of time fumbling over some of the set calculations in the previous version of the `SimpleCSV` 😅 but eventually realized that the now deprecated `_match_x` methods were returning the same set of values. The set syntax greatly improves the readability, I think!